src/main/java/decodes/cwms/CwmsSqlDatabaseIO.java

```diff
@@ -176,11 +181,14 @@ public void connectToDatabase(String sqlDbLocation)
 		
 		try
 		{
-			CwmsConnectionInfo conInfo = 
-				CwmsTimeSeriesDb.getDbConnection(sqlDbLocation, _dbUser, password, dbOfficeId);
-			this.dbOfficeCode = conInfo.getDbOfficeCode();
-			setConnection(conInfo.getConnection());
-			TimeSeriesDb.readVersionInfo(this);
+			conInfo = CwmsTimeSeriesDb.getDbConnection(sqlDbLocation, _dbUser, password, dbOfficeId);
+			openConnections.add(conInfo.getConnection());
+			
+			//TODO Should I set an internal connection object? Shouldn't everybody be calling getConnection()?
```

Comment:
CwmsTimeSeriesDb methods:
- connect() implements base class TimeSeriesDb.connect. This is the method apps call to connect to database.
- getDbConnection() was refactored out a few years ago so that the DECODES interface CwmsSqlDatabaseIO (an older (20 year) interface with a different framework) can use the same static method to actually connect to the DB without duplicating the code there.

As Adam describes to prime the pool, in this new code, it sets the CwmsConnectionInfo (containing login info), sets up the VPD context, checks the db version, etc., and then closes the connection. 

CwmsTimeSeriesDb.getConnection() and freeConnection() are called from DAOs as the application does its business.